src/screens/TransactionViews/TransactionView.res

```diff
@@ -92,25 +88,26 @@ let make = (~entity=TransactionViewTypes.Orders) => {
     }
   }
 
-  let setActiveViewOnLoad = () => {
-    let appliedStatusFilter =
-      filterValueJson->JSON.Encode.object->getDictFromJsonObject->getArrayFromDict("status", [])
+  let settingActiveView = entity => {
+    let customFilterKey = getCustomFilterKey(entity)
+
+    let appliedStatusFilter = filterValueJson->getArrayFromDict(customFilterKey, [])
 
     if appliedStatusFilter->Array.length == 1 {
       let statusValue =
         appliedStatusFilter->getValueFromArray(0, ""->JSON.Encode.string)->JSON.Decode.string
 
       let status = statusValue->Option.getOr("")
-      setActiveView(_ => status->getViewTypeFromString)
+      setActiveView(_ => status->getViewTypeFromString(entity))
     } else {
       setActiveView(_ => All)
     }
   }
 
   React.useEffect(() => {
-    setActiveViewOnLoad()
+    settingActiveView(entity)
     None
-  }, [])
+  }, ())
```

Comment:
Why this line was changed?

Reply:
Because earlier, useEffect runs only once right after component mounts,so the view was not getting set according to values of filters at any point of time.